From c1c85ce74f07275e564854aa1ecd5e229aa1b4a6 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Sun, 28 Jul 2024 17:13:16 +0530
Subject: [PATCH] wifi: ath12k: remove unused members from the HAL rx TLV

The rx_mpdu_start_qcn9274 structure in the HAL RX TLV contains unused
members that occupy 64 bits of space. By removing these unused members,
the size of the HAL RX TLV is reduced from 144 bytes to 136 bytes.

Fixes: (I402ec2a6d3d3) ath12k: subscribe required word mask from rx tlv
Patch-dependency: None

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/hal.c     | 9 +--------
 drivers/net/wireless/ath/ath12k/rx_desc.h | 3 ---
 2 files changed, 1 insertion(+), 11 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/hal.c
+++ b/drivers/net/wireless/ath/ath12k/hal.c
@@ -427,11 +427,6 @@ static void ath12k_hw_qcn9274_rx_desc_co
 	       sizeof(struct rx_msdu_end_qcn9274));
 }
 
-static u32 ath12k_hw_qcn9274_rx_desc_get_mpdu_ppdu_id(struct hal_rx_desc *desc)
-{
-	return __le16_to_cpu(desc->u.qcn9274.mpdu_start.phy_ppdu_id);
-}
-
 static void ath12k_hw_qcn9274_rx_desc_set_msdu_len(struct hal_rx_desc *desc, u16 len)
 {
 	u32 info = __le32_to_cpu(desc->u.qcn9274.msdu_end.info10);
@@ -542,7 +537,7 @@ static u8 ath12k_hal_qcn9274_rx_desc_get
 	return (ipv4 || ipv6);
 }
 
-#define MPDU_START_WMASK 0xFFC
+#define MPDU_START_WMASK 0xFDC
 #define MSDU_END_WMASK 0x137C3
 
 static inline u16 ath12k_hal_rx_mpdu_start_wmask_get(void)
@@ -873,7 +868,6 @@ const struct hal_ops hal_qcn9274_ops = {
 	.rx_desc_get_mpdu_tid = ath12k_hw_qcn9274_rx_desc_get_mpdu_tid,
 	.rx_desc_get_mpdu_peer_id = ath12k_hw_qcn9274_rx_desc_get_mpdu_peer_id,
 	.rx_desc_copy_end_tlv = ath12k_hw_qcn9274_rx_desc_copy_end_tlv,
-	.rx_desc_get_mpdu_ppdu_id = ath12k_hw_qcn9274_rx_desc_get_mpdu_ppdu_id,
 	.rx_desc_set_msdu_len = ath12k_hw_qcn9274_rx_desc_set_msdu_len,
 	.rx_desc_get_msdu_payload = ath12k_hw_qcn9274_rx_desc_get_msdu_payload,
 	.rx_desc_get_mpdu_start_offset = ath12k_hw_qcn9274_rx_desc_get_mpdu_start_offset,
@@ -924,7 +918,6 @@ const struct hal_ops hal_qcn6432_ops = {
 	.rx_desc_get_mpdu_tid = ath12k_hw_qcn9274_rx_desc_get_mpdu_tid,
 	.rx_desc_get_mpdu_peer_id = ath12k_hw_qcn9274_rx_desc_get_mpdu_peer_id,
 	.rx_desc_copy_end_tlv = ath12k_hw_qcn9274_rx_desc_copy_end_tlv,
-	.rx_desc_get_mpdu_ppdu_id = ath12k_hw_qcn9274_rx_desc_get_mpdu_ppdu_id,
 	.rx_desc_set_msdu_len = ath12k_hw_qcn9274_rx_desc_set_msdu_len,
 	.rx_desc_get_msdu_payload = ath12k_hw_qcn9274_rx_desc_get_msdu_payload,
 	.rx_desc_get_mpdu_start_offset = ath12k_hw_qcn9274_rx_desc_get_mpdu_start_offset,
--- a/drivers/net/wireless/ath/ath12k/rx_desc.h
+++ b/drivers/net/wireless/ath/ath12k/rx_desc.h
@@ -122,9 +122,6 @@ struct rx_mpdu_start_qcn9274 {
 	__le32 info1;
 	__le32 pn[4];
 	__le32 info2;
-	__le32 peer_meta_data;
-	__le16 info3;
-	__le16 phy_ppdu_id;
 	__le16 ast_index;
 	__le16 sw_peer_id;
 	__le32 info4;
